python/arcticdb/version_store/library.py

```diff
@@ -104,8 +106,8 @@ def __repr__(self):
 class NameWithDType(NamedTuple):
     """A named tuple. A name and dtype description pair."""
 
-    name: str
-    dtype: str
+    name: Union[Optional[str], List[Optional[str]]]
+    dtype: Union[TypeDescriptor, List[TypeDescriptor]]
```

Comment:
I believe the only reason these need to be `Union[<type>, List[<type>]]` is to handle multi-indexes? In which case I think it would be better to make `index` in the `SymbolDescription` class either `Tuple[NameWithDType]` where the tuple only has one element in the (more common) non-multi-index case, or `Union[NameWithDType, Tuple[NameWithDType]]` so that the most common case is less ugly.